nimare/meta/cbma/mkda.py

```diff
@@ -465,19 +487,34 @@ def _run_fwe_permutation(self, iter_xyz1, iter_xyz2, iter_df1, iter_df2, conn, v
 
         # Generate MA maps and calculate count variables for first dataset
         temp_ma_maps1 = self.kernel_transformer.transform(
-            iter_df1, self.masker, return_type="array"
+            iter_df1, self.masker, return_type="sparse"
         )
         n_selected = temp_ma_maps1.shape[0]
-        n_selected_active_voxels = np.sum(temp_ma_maps1, axis=0)
+        n_selected_active_voxels = temp_ma_maps1.sum(axis=0)
+
+        if isinstance(n_selected_active_voxels, sparse._coo.core.COO):
+            masker = self.masker
+            mask_data = masker.mask_img.get_fdata().astype(bool)
+
+            # Indexing the sparse array is slow, perform masking in the dense array
+            n_selected_active_voxels = n_selected_active_voxels.todense().reshape(-1)
+            n_selected_active_voxels = n_selected_active_voxels[mask_data.reshape(-1)]
+
         del temp_ma_maps1
+        gc.collect()
```

Comment:
Should this be called after every `del`?

Reply:
For MKDAChi2 I had to call this because `del` wasn't freeing up the memory there. I will profile this line to see if it's needed here.

Reply:
Thanks!

Reply:
I'm removing the `gc.collect()`, `del` is working fine here.

Reply:
That is so confusing! I have no clue why it's necessary for some things, but not others... Thanks for checking it in this case though.